pyfqmr/Simplify.pyx

```diff
@@ -36,6 +41,7 @@ cdef class Simplify :
     def __cinit__(self):
         pass
 
+    @cython.boundscheck(False)
```

Comment:
Presumably, we'll now get a segfault or undefined behaviour if you give it an invalid mesh whose faces array contains references to vertices that don't exit? I'm not saying that it wouldn't be worth the speed increase but I would like to know what we're getting ourselves into.

Reply:
This is a good point. I should add some checks for empty arrays. It will do something ugly without that check.

Reply:
I reviewed the code again and I think it should be protected by the size checks. I'll add some automated tests next time I'm working.

Reply:
It's not just empty arrays though right? If I had a setup like:
```python
# 3 random vertices.
vertices = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
# One face referencing a non-existant 12th vertex.
faces = np.array([[0, 1, 12]])
```
then that'd be a segfault too?

Reply:
The arrays are just getting copied so that won't be a problem for this function anyway (though it will be for others). `boundscheck` only checks the copy operation.